ranker/shared/ranker_obj.py

```diff
@@ -162,72 +162,50 @@ def score(self, answer, jaccard_like=False):
+        # Take the max of the property values
+        # This might happen if two KPs have the same underlying data sources
+        # But for some reason return different publication counts        
+        weight_dict = defaultdict(lambda: defaultdict(\
+            lambda: defaultdict( lambda: defaultdict(float))))
         for edge in edges:
-            subject_id, object_id, edge_weight = (
-                edge["subject"],
-                edge["object"],
-                edge["weight"],
-            )
-            subject_index, object_index = index[subject_id], index[object_id]
-            for edge_source in edge_weight.keys():
-                for edge_property in edge_weight[edge_source].keys():
-                    val = edge_weight[edge_source][edge_property]
-                    if edge_source in weight_dict[subject_index][object_index]:
-                        if (
-                            edge_property
-                            in weight_dict[subject_index][object_index][edge_source]
-                        ):
-                            weight_dict[subject_index][object_index][edge_source][
-                                edge_property
-                            ] = max(
-                                weight_dict[subject_index][object_index][edge_source][
-                                    edge_property
-                                ],
-                                val,
-                            )
-                        else:
-                            weight_dict[subject_index][object_index][edge_source][
-                                edge_property
-                            ] = val
-                    else:
-                        weight_dict[subject_index][object_index][edge_source] = {}
-                        weight_dict[subject_index][object_index][edge_source][
-                            edge_property
-                        ] = val
+            subject = edge["subject"]
+            object = edge["object"]
+            edge_weight = edge["weight"]
+
+            for edge_source, edge_properties in edge_weight.items():
+                for edge_property, edge_val in edge_properties.items():
+                    weight_dict[subject][object][edge_source][edge_property] = \
+                        max(weight_dict[subject][object][edge_source][edge_property], edge_val)
 
         qedge_qnode_ids = set(
             [frozenset((e["subject"], e["object"])) for e in self.qedge_by_id.values()]
         )
+
+        # Now go through these edges
+        # Turn each value into an edge weight
+        # Then calculate the graph laplacian
         laplacian = np.zeros((num_nodes, num_nodes))
-        for subject_index in range(num_nodes):
-            q_node_id_subject = node_ids[subject_index][0]
-            for object_id in range(num_nodes):
-                q_node_id_object = node_ids[object_id][0]
+        weight_mat = np.zeros((num_nodes, num_nodes))
```

Comment:
Where do we end up using this weight_mat?

Reply:
We actually don't use it, but boy is it useful for debugging. I would prefer to leave it to aid future investigations.

Reply:
That's cool. Maybe just a comment explaining that then.